File: apps/backend-java/src/main/java/com/housingplatform/features/properties/dto/HostPropertyDetail.java
package com.housingplatform.features.properties.dto;

import com.housingplatform.persistence.enums.AccommodationType;
import com.housingplatform.persistence.enums.BookingMode;
import com.housingplatform.persistence.enums.PropertyStatus;
import java.util.List;
import java.util.UUID;

public record HostPropertyDetail(
    UUID id,
    String title,
    String slug,
    String description,
    AccommodationType propertyType,
    String addressLine1,
    String addressLine2,
    String city,
    String postalCode,
    String district,
    String nearestStationName,
    Integer nearestStationWalkMin,
    PropertyStatus status,
    BookingMode bookingMode,
    int minStayNights,
    List<String> tags,
    Double latitude,
    Double longitude,
    List<UUID> amenityIds,
    List<HostRoomDto> rooms) {}

File: apps/backend-node/src/features/properties/dto/host-property-detail.ts
import type { AccommodationType } from './accommodation-type';
import type { BookingMode } from './booking-mode';
import type { HostRoomDto } from './host-room';
import type { PropertyStatus } from './property-status';

export interface HostPropertyDetail {
  id: string;
  title: string;
  slug: string;
  description: string;
  propertyType: AccommodationType;
  addressLine1: string;
  addressLine2: string | null;
  city: string;
  postalCode: string | null;
  district: string;
  nearestStationName: string | null;
  nearestStationWalkMin: number | null;
  status: PropertyStatus;
  bookingMode: BookingMode;
  minStayNights: number;
  tags: string[];
  latitude: number | null;
  longitude: number | null;
  amenityIds: string[];
  rooms: HostRoomDto[];
}
